presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -300,211 +311,215 @@ public void flushCache()
     }
 
     @Override
-    public Optional<Database> getDatabase(String databaseName)
+    public Optional<Database> getDatabase(MetastoreContext metastoreContext, String databaseName)
     {
-        return get(databaseCache, databaseName);
+        return get(databaseCache, new KeyAndContext<>(getCachingKey(metastoreContext), databaseName));
```

Comment:
Let's make the `getCachingKey` return the `KeyAndContext` that is used as a key for caching